model/softfloat/riscv_softfloat_eq.sail

```diff
@@ -0,0 +1,153 @@
+/*     notice, this list of conditions and the following disclaimer.                     */
+/*  2. Redistributions in binary form must reproduce the above copyright                 */
+/*     notice, this list of conditions and the following disclaimer in                   */
+/*     the documentation and/or other materials provided with the                        */
+/*     distribution.                                                                     */
+/*                                                                                       */
+/*  THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS''                   */
+/*  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED                    */
+/*  TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A                      */
+/*  PARTICULAR PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR                  */
+/*  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,                         */
+/*  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT                     */
+/*  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF                     */
+/*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND                  */
+/*  ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,                   */
+/*  OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT                   */
+/*  OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF                   */
+/*  SUCH DAMAGE.                                                                         */
+/*=======================================================================================*/
+
+/* **************************************************************** */
+/* This file implements the floating-point for equal and consumes   */
+/* by the softfloat API contract, riscv_softfloat_interface.sail.   */
+/*                                                                  */
+/* The below floating-point types are supported.                    */
+/* 1. Double-precision, aka 64 bits floating-point.                 */
+/* 2. Single-precision, aka 32 bits floating-point.                 */
+/* 3. Half-precision, aka 16 bits floating-point.                   */
+/*                                                                  */
+/* However, the implementation still need some global variables in  */
+/* file riscv_softfloat_interface.sail. It includes:                */
+/* 1. float_result.                                                 */
+/* 2. float_fflags.                                                 */
+/* 3. float_roundingMode.                                           */
+/* **************************************************************** */
+
+val            operator == : (bits_S, bits_S) -> bool
+val            operator |  : (bool, bool) -> bool
```

Comment:
Are these really necessary?

Reply:
May be not, let me double-check if there is some build error without this.